libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1103,233 +1250,282 @@ Vector3f AC_PosControl::lean_angles_to_accel_NEU_cmss(const Vector3f& att_target
-/// this is used to initiate the offsets when initialise the position controller or do an offset reset
+// Initializes vertical (U) offsets to match their respective targets.
 void AC_PosControl::init_offsets_U()
 {
     // check for offset target timeout
     uint32_t now_ms = AP_HAL::millis();
     if (now_ms - _posvelaccel_offset_target_u_ms > POSCONTROL_POSVELACCEL_OFFSET_TARGET_TIMEOUT_MS) {
-        _pos_offset_target_neu_cm.z = 0.0;
-        _vel_offset_target_neu_cms.z = 0.0;
-        _accel_offset_target_neu_cmss.z = 0.0;
+        _pos_offset_target_neu_m.z = 0.0;
+        _vel_offset_target_neu_ms.z = 0.0;
+        _accel_offset_target_neu_mss.z = 0.0;
     }
+
     // set position offset to target
-    _pos_offset_neu_cm.z = _pos_offset_target_neu_cm.z;
+    _pos_offset_neu_m.z = _pos_offset_target_neu_m.z;
 
     // set velocity offset to target
-    _vel_offset_neu_cms.z = _vel_offset_target_neu_cms.z;
+    _vel_offset_neu_ms.z = _vel_offset_target_neu_ms.z;
 
     // set acceleration offset to target
-    _accel_offset_neu_cmss.z = _accel_offset_target_neu_cmss.z;
+    _accel_offset_neu_mss.z = _accel_offset_target_neu_mss.z;
 }
 
 #if AP_SCRIPTING_ENABLED
-// add an additional offset to vehicle's target position, velocity and acceleration
-// units are m, m/s and m/s/s in NED frame
-// Z-axis is not currently supported and is ignored
-bool AC_PosControl::set_posvelaccel_offset(const Vector3f &pos_offset_NED, const Vector3f &vel_offset_NED, const Vector3f &accel_offset_NED)
-{
-    set_posvelaccel_offset_target_NE_cm(pos_offset_NED.topostype().xy() * 100.0, vel_offset_NED.xy() * 100.0, accel_offset_NED.xy() * 100.0);
-    set_posvelaccel_offset_target_U_cm(-pos_offset_NED.topostype().z * 100.0, -vel_offset_NED.z * 100, -accel_offset_NED.z * 100.0);
+// Sets additional position, velocity, and acceleration offsets in meters (NED frame) for scripting.
+// Offsets are added to the controller’s internal target.
+// Used in LUA
+bool AC_PosControl::set_posvelaccel_offset(const Vector3f &pos_offset_NED_m, const Vector3f &vel_offset_NED_ms, const Vector3f &accel_offset_NED_mss)
+{
+    // Convert NED inputs to NEU frame: Z is inverted
+    set_posvelaccel_offset_target_NE_m(pos_offset_NED_m.topostype().xy(), vel_offset_NED_ms.xy(), accel_offset_NED_mss.xy());
+    set_posvelaccel_offset_target_U_m(-pos_offset_NED_m.topostype().z, -vel_offset_NED_ms.z, -accel_offset_NED_mss.z);
     return true;
 }
 
-// get position and velocity offset to vehicle's target velocity and acceleration
-// units are m and m/s in NED frame
-bool AC_PosControl::get_posvelaccel_offset(Vector3f &pos_offset_NED, Vector3f &vel_offset_NED, Vector3f &accel_offset_NED)
+// Retrieves current scripted offsets in meters (NED frame).
+// Used in LUA
+bool AC_PosControl::get_posvelaccel_offset(Vector3f &pos_offset_NED_m, Vector3f &vel_offset_NED_m, Vector3f &accel_offset_NED_m)
```

Comment:
it might make sense to change the extensions for the velocity and accel offset arguments to _ms and _mss respectively.  same goes for the header file definition of this method.  again, not a blocker